src/docs/product/integrations/integration-platform/ui-components/alert-rule-action.mdx

```diff
@@ -0,0 +1,140 @@
+---
+title: Alert Rule Action
+sidebar_order: 3
+description: "Learn about the alert rule action UI component."
+---
+
+The Alert Rule Action component allows the developer to get parameters to define routing or configuration in alert rules for their service. When alert rules are triggered for a user, the configured service will receive [Issue Alert](/product/integrations/integration-platform/webhooks/#issue-alerts) and [Metric Alert](/product/integrations/integration-platform/webhooks/#metric-alerts) webhook events with the specified settings for that service to further route the alert.
+
+![Example of an alert rule UI component.](alert-rule-component-action-example.png)
+
+![Example of an alert rule settings modal.](alert-rule-component-modal-example.png)
+
+## Schema
+
+```javascript
+ {
+    "type": "alert-rule-action",
+    "title": <String>,
+    "settings": {
+        "type": "alert-rule-settings",
+        "uri": <URI>,
+        "required_fields": <Array<FormField>>,
+        "optional_fields": <Array<FormField>>,
+        "description": <String>
+    }
+}
+```
+
+## Attributes
+
+- `title` - (Required) The title shown in the UI component.
+- `uri` - (Required) Sentry will make a POST request to the URI when the User submits the form. If the services fails to process the request (status code >= 400), this component will bubble up the error to the User with the provided response text.
+- `required_fields` - (Required) List of [FormField](/product/integrations/integration-platform/ui-components/formfield) components the User is required to complete.
+- `optional_fields` - (Optional) List of [FormField](/product/integrations/integration-platform/ui-components/formfield) components the User may complete.
+- `description` - (Optional) Text that will be displayed above the form. Limited to 140 characters.
+
+## Example
+
+```javascript
+
+"elements": [
+    {
+        "type": "alert-rule-action",
+        "title": "Send Alerts to",
+        "settings": {
+        "type": "alert-rule-settings",
+        "uri": "/sentry/alert-rule",
+            "required_fields": [
+                {
+                    "type": "text",
+                    "label": "Channel",
+                    "name": "channel"
+                }
+            ]
+        }
+    }
+]
+```
+
+
+## Issue Alert Request Format
+
+When an issue alert fires, your service will need to read the settings from the alert payload. The `settings` are in `data.issue_alert.settings`. See the full [Issue Alert webhook documentation](/product/integrations/integration-platform/webhooks/#issue-alerts) for more information.
+
+### Payload
+
+```
```

Comment:
This should also be `json`, and more generally, in markdown it just dumps whatever you have written into the page, and most editors don't have smart configs for editing markdown to prettify it before hand, so you gotta manually fix spacing issues and stuff. 

This is technically fine spacing-wise, but I'd prefer 2-space indents since it's so heavily nested. IMO 2-space indents are way better than 4-space for docs since it's more readible when the window is thinner or half-screened.